optimizely/odp/zaius_graphql_api_manager.py

```diff
@@ -0,0 +1,203 @@
+ \"has_email_opted_in\", \"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+
+    query MyQuery {
+    customer(vuid: "d66a9d81923d4d2f99d8f64338976322") {
+     audiences(subset:["has_email", "has_email_opted_in", "push_on_sale"]) {
+       edges {
+         node {
+           name
+           state
+         }
+       }
+     }
+    }
+    }
+
+
+    [GraphQL Response]
+    {
+    "data": {
+     "customer": {
+       "audiences": {
+         "edges": [
+           {
+             "node": {
+               "name": "has_email",
+               "state": "qualified",
+             }
+           },
+           {
+             "node": {
+               "name": "has_email_opted_in",
+               "state": "qualified",
+             }
+           },
+            ...
+         ]
+       }
+     }
+    }
+    }
+
+    [GraphQL Error Response]
+    {
+    "errors": [
+     {
+       "message": "Exception while fetching data (/customer) : java.lang.RuntimeException:
+       could not resolve _fs_user_id = asdsdaddddd",
+       "locations": [
+         {
+           "line": 2,
+           "column": 3
+         }
+       ],
+       "path": [
+         "customer"
+       ],
+       "extensions": {
+         "classification": "InvalidIdentifierException"
+       }
+     }
+    ],
+    "data": {
+     "customer": null
+    }
+    }
+"""
+
+
+REQUEST_TIMEOUT: Final = 10
+
+
+class ZaiusGraphQLApiManager:
+    """Interface for manging the fetching of audience segments."""
+    def __init__(self, logger: Optional[optimizely_logger.Logger] = None):
+        self.logger = logger or optimizely_logger.NoOpLogger()
+
+    def fetch_segments(self, api_key: str, api_host: str, user_key: str,
+                       user_value: str, segments_to_check: list[str]) -> Optional[list[str]]:
+        """
+        Fetch segments from ODP GraphQL API.
+
+        Args:
+            api_key: public api key
+            api_host: domain url of the host
+            user_key: vuid or fs_user_id (client device id or fullstack id)
+            user_value: vaue of user_key
+            segments_to_check: lit of segments to check
+
+        Returns:
+            Audience segments from GraphQL.
+        """
+        url = f'{api_host}/v3/graphql'
+        request_headers = {'content-type': 'application/json',
+                           'x-api-key': str(api_key)}
+
+        segments_filter = self.make_subset_filter(segments_to_check)
+        payload_dict = {
+            'query': 'query {customer(' + str(user_key) + ': "' + str(user_value) + '") '
+                     '{audiences' + segments_filter + ' {edges {node {name state}}}}}'
+        }
+
+        try:
+            response = requests.post(url=url,
+                                     headers=request_headers,
+                                     data=json.dumps(payload_dict),
+                                     timeout=REQUEST_TIMEOUT)
+
+            response.raise_for_status()
```

Comment:
If you comment out this line, all tests still pass. I think we need to get the tests a little closer to the real thing.
We should return a `requests.Response` object from `requests.post` in all tests where possible.

Reply:
Good catch. Fixed.